src/Features/CSharp/Portable/CodeRefactorings/InvertIf/InvertIfCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,218 @@
+        {
+            var ifStatement = token.GetAncestor<IfStatementSyntax>();
+            if (ifStatement == null || ifStatement.Else == null)
+            {
+                return null;
+            }
+
+            // Tweak 
+            var span = TextSpan.FromBounds(ifStatement.GetFirstToken().Span.Start, ifStatement.CloseParenToken.Span.End);
+            if (!span.IntersectsWith(textSpan))
+            {
+                return null;
+            }
+
+            return ifStatement;
+        }
+
+        protected override SyntaxNode GetRootWithInvertIfStatement(Workspace workspace, SemanticModel model, SyntaxNode ifStatement, CancellationToken cancellationToken)
+        {
+            var ifNode = (IfStatementSyntax)ifStatement;
+
+            // In the case that the else clause is actually an else if clause, place the if
+            // statement to be moved in a new block in order to make sure that the else
+            // statement matches the right if statement after the edit.
+            var newIfNodeStatement = ifNode.Else.Statement.Kind() == SyntaxKind.IfStatement
+                ? SyntaxFactory.Block(ifNode.Else.Statement)
+                : ifNode.Else.Statement;
+
+            ifNode = ifNode.WithCondition(Negate(ifNode.Condition, model, cancellationToken))
+                .WithStatement(newIfNodeStatement)
+                .WithElse(ifNode.Else.WithStatement(ifNode.Statement))
+                .WithAdditionalAnnotations(Formatter.Annotation);
+
+            // get new root
+            return model.SyntaxTree.GetRoot().ReplaceNode(ifStatement, ifNode);
+        }
+
+        private bool TryNegateBinaryComparisonExpression(
+            ExpressionSyntax expression,
+            SemanticModel semanticModel,
+            CancellationToken cancellationToken,
+            out ExpressionSyntax result)
+        {
+            if (s_binaryMap.TryGetValue(expression.Kind(), out var negatedExpressionInfo))
+            {
+                var binaryExpression = (BinaryExpressionSyntax)expression;
+                var (negatedExpressionType, negatedOperatorType) = negatedExpressionInfo;
+
+                // Certain expressions can never be negative, such as length or unsigned numeric types.
+                // If these expressions are compared with zero using <, >, or =, we construct the negated
+                // binary expression to reflect that it will never be negative.
+                // For example, the expression Array.Length > 0, becomes Array.Length == 0 when negated.
+                var operation = semanticModel.GetOperation(binaryExpression);
+                if (IsSpecialCaseBinaryExpression(operation as IBinaryOperation, cancellationToken))
+                {
+                    negatedOperatorType = binaryExpression.OperatorToken.Kind() == SyntaxKind.EqualsEqualsToken
+                        ? binaryExpression.Right is LiteralExpressionSyntax ? SyntaxKind.GreaterThanToken : SyntaxKind.LessThanToken
```

Comment:
> binaryExpression.Right is LiteralExpressionSyntax

Instead of checking for literal syntax, consider checking for `IOperation.ConstantValue.HasValue`.